app/services/gmail/download.py

```
@@ -133,24 +133,23 @@ def callback(
+                progress=progress,
+                message=f"Fetched {fetched}/{total_emails} emails...",
             )
 
             # Rate limiting: sleep every 5 batches to avoid hitting API limits
             if (i // batch_size + 1) % 5 == 0:
                 time.sleep(0.3)
 
     except Exception as e:
-        state.download_status["done"] = True
-        state.download_status["error"] = f"Error fetching emails: {str(e)}"
+        state.update_download_status(done=True, error=f"Error fetching emails: {str(e)}")
```

Ruff (0.14.10)

Use explicit conversion flag

Replace with conversion flag

(RUF010)